lib/trino-filesystem-alluxio/src/main/java/io/trino/filesystem/alluxio/AlluxioFileSystem.java

```diff
@@ -0,0 +1,380 @@
+    }
+
+    @Override
+    public TrinoOutputFile newOutputFile(Location location)
+    {
+        ensureNotRootLocation(location);
+        ensureNotEndWithSlash(location);
+        return new AlluxioFileSystemOutputFile(rootLocation, location, fileSystem, mountRoot);
+    }
+
+    @Override
+    public void deleteFile(Location location)
+            throws IOException
+    {
+        ensureNotRootLocation(location);
+        ensureNotEndWithSlash(location);
+        try {
+            fileSystem.delete(convertToAlluxioURI(location, mountRoot));
+        }
+        catch (FileDoesNotExistException e) {
+        }
+        catch (AlluxioException e) {
+            throw new IOException("Error deleteFile %s".formatted(location), e);
+        }
+    }
+
+    @Override
+    public void deleteDirectory(Location location)
+            throws IOException
+    {
+        try {
+            AlluxioURI uri = convertToAlluxioURI(location, mountRoot);
+            URIStatus status = fileSystem.getStatus(uri);
+            if (status == null) {
+                return;
+            }
+            if (!status.isFolder()) {
+                throw new IOException("delete directory cannot be called on a file %s".formatted(location));
+            }
+            DeletePOptions deletePOptions = DeletePOptions.newBuilder().setRecursive(true).build();
+            // recursive delete on the root directory must be handled manually
+            if (location.path().isEmpty() || location.path().equals(mountRoot)) {
+                for (URIStatus uriStatus : fileSystem.listStatus(uri)) {
+                    fileSystem.delete(new AlluxioURI(uriStatus.getPath()), deletePOptions);
+                }
+            }
+            else {
+                fileSystem.delete(uri, deletePOptions);
+            }
+        }
+        catch (FileDoesNotExistException | NotFoundRuntimeException e) {
+        }
+        catch (AlluxioException e) {
+            throw new IOException("Error deleteDirectory %s".formatted(location), e);
+        }
+    }
+
+    @Override
+    public void renameFile(Location source, Location target)
+            throws IOException
+    {
+        try {
+            ensureNotRootLocation(source);
+            ensureNotEndWithSlash(source);
+            ensureNotRootLocation(target);
+            ensureNotEndWithSlash(target);
+        }
+        catch (IllegalStateException e) {
+            throw new IllegalStateException("Cannot rename file from %s to %s as one of them is root location"
+                    .formatted(source, target), e);
+        }
+        AlluxioURI sourceUri = convertToAlluxioURI(source, mountRoot);
+        AlluxioURI targetUri = convertToAlluxioURI(target, mountRoot);
+
+        try {
+            if (!fileSystem.exists(sourceUri)) {
+                throw new IOException("Cannot rename file %s to %s as file %s doesn't exist"
+                        .formatted(source, target, source));
```

Comment:
General formatting comment for all the code:
We either put every argument on a newline or all on same line

Reply:
formatted